### PR TITLE
Add downgrade workflow, fix lower bounds

### PR DESCRIPTION
**Pull request by Sbozzolo**
This commit adds the downgrade github action and cleans up some incorrect lower bounds. I also removed a couple of packages that could be trivially removed.

Closes #1781

### CODE REVIEW

**Comment by charleskawczynski** [comment]
I think we can split this PR up into:

 - Qualifying packages (`using ...` --> `import ...`)
 - Removing unused packages (`Combinatorics`)
 - Downgrade GHA

I'm happy to help peel off some pieces

**Comment by Sbozzolo** [comment]
Removing some of the packages was needed to add the downgrade job, otherwise it was impossible to solve the version constraint (see some of the failed github actions). 

Also, the job is still failing, so I haven't fully figured out what needs to be done.

I'd like to merge this (fixing the lower bounds) before making the new release, if possible. The lower bounds of the compats are wrong, which makes adding downgrade to other packges harder.

**Comment by charleskawczynski** [comment]
> I'd like to merge this (fixing the lower bounds) before making the new release, if possible. The lower bounds of the compats are wrong, which makes adding downgrade to other packges harder.

I would like to merge this too, but I think I'd prefer we don't wait for the next release. I was hoping to make a release today. We can always make another quick patch release soon after. I'd like to have frequent release cycles when we have frequent delivered features/improvements, and there are a bunch in main right now.

**Comment by Sbozzolo** [comment]
> > I'd like to merge this (fixing the lower bounds) before making the new release, if possible. The lower bounds of the compats are wrong, which makes adding downgrade to other packges harder.

I think I can fix this in the next hour or so.

**Comment by Sbozzolo** [comment]
The downgrade tests passed. I am rebasing and this should be ready.

File: NEWS.md [hunk 1]
```diff
@@ -19,6 +19,13 @@ main
  - We've refactored some modules to use less internals. PR [#2053](https://github.com/CliMA/ClimaCore.jl/pull/2053), PR [#2052](https://github.com/CliMA/ClimaCore.jl/pull/2052), [#2051](https://github.com/CliMA/ClimaCore.jl/pull/2051), [#2049](https://github.com/CliMA/ClimaCore.jl/pull/2049).
  - Some work was done in attempt to reduce specializations and compile time. PR [#2042](https://github.com/CliMA/ClimaCore.jl/pull/2042), [#2041](https://github.com/CliMA/ClimaCore.jl/pull/2041)
 
+### ![][badge-🐛bugfix] Fix lower compat bounds
+
+`ClimaCore` was falsely with certain versions of packages, while that was not
+really true. PR [#2078](https://github.com/CliMA/ClimaCore.jl/pull/2078) fixes
+the lower bounds and adds a GitHub Action workflow to test it. `ClimaCore` now
+requires Julia 1.10 or greater.
```
Comment (charleskawczynski) [inline NEWS.md:27]:
```suggestion
`ClimaCore` had incorrect lower compat bounds.
PR [#2078](https://github.com/CliMA/ClimaCore.jl/pull/2078) fixes the lower bounds and adds a GitHub
Action workflow to test it. `ClimaCore` now requires
Julia 1.10 or greater.
```

**Review by charleskawczynski — APPROVED** [review]

**Comment by charleskawczynski** [comment]
Ah, sorry, I didn't realize that #2080 would result in conflicts. I hope it's not too complicated to fix. I support merging this once it's resolved / rebased.

**Pull request merged**